test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,242 @@
+
+std::string generate_fastq_file(size_t const entries_size)
+{
+    std::ostringstream stream_buffer{};
+    seqan3::debug_stream_type format_stream{stream_buffer};
+
+    std::string const id{"@name"};
+
+    auto seed = default_seed;
+
+    for (size_t i = 0; i < entries_size; ++i, ++seed)
+    {
+        auto random_sequence = seqan3::test::generate_sequence<seqan3::dna5>(default_sequence_length, 0, seed);
+        auto random_qualties = seqan3::test::generate_sequence<seqan3::phred42>(default_sequence_length, 0, seed);
+        format_stream << id << '\n'  // write id
+                      << random_sequence << '\n' // write sequence
+                      << '+' << '\n' // write qualities separator
+                      << random_qualties << '\n'; // write qualities
+    }
+
+    return stream_buffer.str();
+}
+
+// ============================================================================
+// save file on disc temporarily
+// ============================================================================
+
+auto create_fastq_file(size_t const entries)
+{
+    // create temporary file, automatically removed on destruction
+    seqan3::test::tmp_filename fastq_file{"format_fastq_benchmark_test_file.fastq"};
+    auto fastq_file_path = fastq_file.get_path();
+
+    // fill temporary file with a fastq file
+    std::ofstream ostream{fastq_file_path};
+    ostream << generate_fastq_file(entries);
+    ostream.close();
+    return fastq_file;
+}
+
+// ============================================================================
+// write 3-line entry to stream as often as possible
+// ============================================================================
+
+void fastq_write_to_stream_seqan3(benchmark::State & state)
+{
+    std::ostringstream ostream;
+    seqan3::sequence_file_output fout{ostream, seqan3::format_fastq{}};
+
+    std::string const id{"@name"};
+    auto seq = seqan3::test::generate_sequence<seqan3::dna5>(default_sequence_length, 0, default_seed);
+    auto qual = seqan3::test::generate_sequence<seqan3::phred42>(default_sequence_length, 0, default_seed);
+
+    for (auto _ : state)
+        fout.emplace_back(seq, id, qual);
+}
+
+// ============================================================================
+// seqan3 benchmark
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// read dummy fastq file from a stream
+// ----------------------------------------------------------------------------
+
+void fastq_read_from_stream_seqan3(benchmark::State & state)
+{
+    std::string file = generate_fastq_file(state.range(0));
+
+    seqan3::dna5_vector sequence{};
+    std::string id{};
+    std::vector<seqan3::phred42> quality{};
+
+    for (auto _ : state)
+    {
+        state.PauseTiming();
+        std::istringstream istream{file};
```

Comment:
Why don't we move this line before the loop, after `std::string file` is created? 
`file` does not change in this function and I do not see why we should create new istreams all the time.

Reply:
because it will be consumed after the first execution of the benchmark since it is an input stream. Hence calling it again without resetting it before will give a wrong benchmark. 
I could try to set the underlying buffer, which might be a non-invasive operation, but if handling it inside of a paused time block has no time effects on the benchmark the more complicated code seems not worth the effort.

Reply:
Okay, then it's fine!